partner_credit_limit/models/sale.py

```diff
@@ -0,0 +1,68 @@
+# See LICENSE file for full copyright and licensing details.
+
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def check_limit(self):
+        self.ensure_one()
+        partner = self.partner_id
+        user_id = self.env['res.users'].search([
+            ('partner_id', '=', partner.id)], limit=1)
+        if user_id and not user_id.has_group('base.group_portal') or not \
+                user_id:
+            moveline_obj = self.env['account.move.line']
+            movelines = moveline_obj.search(
+                [('partner_id', '=', partner.id),
+                 ('account_id.account_type', 'in',
+                    ['Receivable', 'Payable']),
+                 ('parent_state','!=','cancel')]
+            )
+            confirm_sale_order = self.search(
+                [('partner_id', '=', partner.id),
+                 ('state', '=', 'sale'),
+                 ('invoice_status', '!=', 'invoiced')])
+            debit, credit = 0.0, 0.0
+            amount_total = 0.0
+            for status in confirm_sale_order:
+                amount_total += status.amount_total
```

Comment:
Use inline code:
amount_total = sum(sale.amount_total for sale in confirm_sale_order)

Reply:
Use inline code:
amount_total = sum(confirm_sale_order.mapped('amount_total')))